index.bs

```diff
@@ -2367,43 +2367,31 @@ To <dfn>get the browsing context info</dfn> given |context|,
 
    Note: This includes the fragment component of the URL.
 
-1. Let |child info| be the result of [=get the descendent browsing contexts=]
-   given |context id|, |depth| + 1, and |max depth|.
+1. If |max depth| is null, or |max depth| is greater than 0:
 
-1. Let |context info| be a [=map=] matching the
-   <code>BrowsingContextInfo</code> production with the <code>context</code>
-   field set to |context id|, the <code>parent</code> field set to |parent id|
-   if |depth| is 0, or unset otherwise, the <code>url</code> field set to
-   |url|, and the <code>children</code> field set to |child info|.
-
-1. Return |context info|.
-
-</div>
-
-<div algorithm>
-To <dfn>get the descendent browsing contexts</dfn> given |parent id|, |depth|
-and |max depth|:
+  1. Let |child infos| be an empty list.
 
-1. If |max depth| is greater than zero, and |depth| is equal to |max depth|,
-   return null.
+  1. Let |child contexts| be a list containing all [=/browsing contexts=]
+     which are [=child browsing contexts=] of |parent|.
 
-1. Let |parent| be the result of [=trying=] to [=get a browsing context=]
-   given |parent id|.
+  1. Let |child depth| be |max depth| - 1 if |max depth| is not null, or null otherwise.
 
-1. If |parent| is null, let |child contexts| be a list containing all [=top-level
-   browsing contexts=]. Otherwise let |child contexts| be a list containing all
-   [=/browsing contexts=] which are [=child browsing contexts=] of |parent|.
+  1. For each |context| of |child contexts|:
 
-1. Let |contexts info| be a list.
+    1. Let |info| be the result of [=get the browsing context info=] given
+       |context|, |child depth|, and false.
 
-1. For each |context| of |child contexts|:
+    1. Append |info| to |child infos|
 
-  1. Let |info| be the result of [=get the browsing context info=] given
-     |context|, |depth|, and |max depth|.
+1. Otherwise let |child infos| be null.
```

Comment:
We're a bit inconsistent in whether "otherwise" clauses should be part of the same list item as the corresponding "if" or not; I tend to think they probably should. But in this case it would perhaps be clearer to do:
```
1. Let |child infos| be null.

1. If |max depth| is null, or |max depth| is greater than 0:

 1. Set |child infos| to an empty list
 […]
```

Reply:
Will be done in the next update that i'm going to push in a minute.